src/components/LocationSearch/LocationSearch.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { useCombobox } from 'downshift';
+
+import usePlacesAutocomplete from './usePlacesAutocomplete';
+import styles from './location-search.module.css';
+import poweredByGoogle from './powered_by_google.png';
+
+const LocationSearch = ({ onSelectedItemChange }) => {
+  const [query, setQuery] = React.useState('');
+
+  const { places, getPlaceById } = usePlacesAutocomplete(query);
+
+  const handleSelectedItemChange = async ({ selectedItem }) => {
+    if (!selectedItem) {
+      return;
+    }
+
+    const result = await getPlaceById(selectedItem.placeId);
+    const { lat, lng } = result[0].geometry.location;
+
+    onSelectedItemChange({ lat: lat(), lng: lng() });
+  };
+
+  const {
+    isOpen,
+    getLabelProps,
+    setInputValue,
+    getMenuProps,
+    getInputProps,
+    getComboboxProps,
+    highlightedIndex,
+    getItemProps,
+  } = useCombobox({
+    items: places,
+    onInputValueChange: ({ inputValue }) => setQuery(inputValue),
+    itemToString: (item) => item.label + ', ' + item.subLabel,
+    onSelectedItemChange: handleSelectedItemChange,
+  });
+
+  return (
+    <>
+      <Helmet>
+        <script
+          src={`https://maps.googleapis.com/maps/api/js?key=${process.env.REACT_APP_GOOGLE_MAPS_API_KEY}&libraries=places`}
+        />
+      </Helmet>
+
+      <div className={styles.wrapper}>
+        <label className={styles.label} {...getLabelProps()}>
+          Search for a location
+        </label>
+
+        <div className={styles.controls} {...getComboboxProps()}>
+          <input
+            onFocus={() => setInputValue('')}
+            placeholder="Search for a location"
+            className={styles.input}
+            style={{
+              borderRadius: isOpen ? '5px 5px 0 0' : 5,
+              borderBottom: isOpen ? '1px solid #ccc' : 'none',
+            }}
+            {...getInputProps()}
+          />
+        </div>
+
+        <div {...getMenuProps()}>
```

Comment:
* `getMenuProps` will apply `role="listbox"`
* `getItemProps` will apply `role="option"`

I think that the `option` role is required to be a direct descendant of `listbox`.

I'd recommend pairing this with a list html, like so:
```jsx
<ul {...getMenuProps()}>
  <li {...getItemProps({ item, index })} />
</ul>
```

Reply:
Great spot, thank you 🙇